init-odkx-sync-endpoint.py

```diff
@@ -82,38 +131,38 @@ def run_interactive_config():
 
         if not manual_certificate:
-            os.system("sudo certbot certonly --standalone \
-            --email {} \
-            -d {} \
-            --rsa-key-size 4096 \
-            --agree-tos \
-            --cert-name bootstrap \
-            --keep-until-expiring \
-            --non-interactive".format(env["HTTPS_ADMIN_EMAIL"], env["HTTPS_DOMAIN"]))
+            skip_setup_certificate(env)
         else:
-            cert_fullchain_path: str = env.get("CERT_FULLCHAIN_PATH")
-            cert_privkey_path: str = env.get("CERT_PRIVKEY_PATH")
-            typer.echo('Please enter path to fullchain .pem/.crt file')
-            cert_fullchain_path = typer.prompt(f"fullchain file [({cert_fullchain_path})]", default=cert_fullchain_path, show_default=False)
-            typer.echo('Please enter path to private key .pem file')
-            cert_privkey_path = typer.prompt(f"private key file [({cert_privkey_path})]", default=cert_privkey_path, show_default=False)
-            if not cert_fullchain_path or not cert_privkey_path:
-                typer.echo('Input not provided, re-run this script with correct inputs.')
-                raise typer.Exit()
-            # Compute absolute paths from relative path inputs
-            cert_fullchain_path = path.abspath(cert_fullchain_path)
-            cert_privkey_path = path.abspath(cert_privkey_path)
-            if not path.exists(cert_fullchain_path) or not path.exists(cert_privkey_path):
-                typer.echo('File at the given path do not exists, re-run this script with correct inputs.')
-                raise typer.Exit()
-            env['CERT_FULLCHAIN_PATH'] = cert_fullchain_path
-            env['CERT_PRIVKEY_PATH'] = cert_privkey_path
-
+            setup_certificate(env)
+        
+        if allow_save_progress:
+            save_progress('manual_certificate', manual_certificate)
         typer.echo("Attempting to save updated https configuration")
-        write_to_env_file(env_file_location, env)
-
-    return (enforce_https, env)
-
+    return enforce_https
+
+def is_cache_present() -> bool:
+    return os.path.exists('progress.json') and os.stat('progress.json').st_size != 0 
+
+def save_progress(key: str, value: Union[str, bool, Dict[str, str]]):
+    if is_cache_present():
+        with open('progress.json', 'r') as progress_file:
+            data: Dict[str,  Union[str, bool, Dict[str, str]]] = json.load(progress_file) 
+            data.update({key: value})
+    else :
+        data: Dict[str,  Union[str, bool, Dict[str, str]]] = {key: value}
+
+    with open('progress.json', 'w') as progress_file:
+        json.dump(data, progress_file)
+
+def load_progress() -> Dict[str,  Union[str, bool, Dict[str, str]]]:
+    with open('progress.json', 'r') as progress_file:
+        data: Dict[str,  Union[str, bool, Dict[str, str]]] = json.load(progress_file)
+        if ('env' not in data.keys() or 'enforce_https' not in data.keys() and 'manual_certificate' not in data.keys()):
```

Comment:
When chaining ands and ors, it's usually good to add parenthesis to explicitly encode the order of operations you intend.